root/components/Aliases.js

```diff
@@ -0,0 +1,105 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+const React = require('react');
+const Frag = require('./Frag');
+const {l} = require('../static/scripts/common/i18n');
+const isolateText = require('../static/scripts/common/utility/isolateText');
+const formatDate = require('../static/scripts/common/utility/formatDate');
+const getLanguageName = require('../static/scripts/common/utility/getLanguageName');
+const DocLink = require('../static/scripts/common/components/DocLink');
+const EntityLink = require('../static/scripts/common/components/EntityLink');
+
+const buildAliasRows = (aliases, disallowEditing, entity) => {
+  const aliasRows = [];
+  for (let i = 0; i < aliases.length; i++) {
+    const alias = aliases[i];
+    const showSortName = alias.name !== alias.sort_name;
+    let actions = null;
+    if ($c.user_exists && !disallowEditing) {
+      actions = (
+        <Frag>
+          <a href={`/${entity.entityType}/${entity.gid}/alias/${alias.id}/edit`}>
+            {l('Edit')}
+          </a>
+          {' | '}
+          <a href={`/${entity.entityType}/${entity.gid}/alias/${alias.id}/delete`}>
+            {l('Remove')}
+          </a>
+        </Frag>
+      );
+    }
+    const row = (
+      <tr className={i % 2 ? 'even' : 'odd'} key={i}>
+        <td colSpan={showSortName ? 1 : 2}>
+          {alias.edits_pending ? <span className="mp">{isolateText(alias.name)}</span> : isolateText(alias.name)}
+        </td>
+        {showSortName ? <td>{isolateText(alias.sort_name)}</td> : null}
+        <td>{formatDate(alias.begin_date)}</td>
+        <td>{alias.ended ? alias.end_date ? formatDate(alias.end_date) : l('Unknown') : null}</td>
+        <td>{alias.typeName}</td>
+        <td>
+          {alias.locale ? getLanguageName(alias.locale) : null}
+          {alias.primary_for_locale ? l(' ({primary})', {__react: true, primary: <span className="comment">{l('primary')}</span>}) : null }
```

Comment:
Not sure how to do this with bracketed? It does not support react, it felt it was easier to do it this way.

Reply:
You should be able to tweak it to allow passing additional args to `l`:

```JavaScript
function bracketed(text, args) {
  if (text) {
    return l(' ({text})', {text, ...args});
  }
  return '';
}

bracketed(<span className="comment">{l('primary')}</span>, {__react: true});
```

It's just nice if all the usages are consistent in case we want to change something later.